src/nut_agent.cc

```diff
@@ -246,13 +262,48 @@ void NUTAgent::advertisePhysics ()
             }
         }
 
+        // send alarms as bitmap
+        bool has_alarms = false;
+        if (device.second.hasProperty ("ups.alarm")) {
+            const auto &alarms = device.second.property ("ups.alarm");
+            uint16_t bitfield = 0;
+            int bit = 0;
+            for (const auto &i : alarmsList) {
+                if (alarms.find(i)) {
+                    bitfield |= (1 << bit);
+                    has_alarms = true;
+                }
+                bit++;
+            }
+            zmsg_t *msg = fty_proto_encode_metric (
+                NULL,
+                time (NULL),
+                _ttl,
+                "ups.alarm",
+                device.second.assetName ().c_str (),
+                std::to_string (bitfield).c_str (),
+                "");
+            if (msg) {
+                log_debug ("sending new alarms for element_src = '%s', value = '%s' (%s)",
```

Comment:
this is more a measurement/metric than an alarm.
"sending new measurement ups.alarm .." seems more appropiate